source/slang/slang-ast-type.h

```diff
@@ -678,21 +691,59 @@ class FuncType : public Type
 };
 
 // A tuple is a product of its member types
-class TupleType : public Type
+class TupleType : public DeclRefType
 {
     SLANG_AST_CLASS(TupleType)
 
-    // Construct a unary tupletion
-    TupleType(ArrayView<Type*> memberTypes)
+    Index getMemberCount() const;
+    Type* getMember(Index i) const;
+    Type* getTypePack() const;
+
+};
```

Comment:
Yes, tuple types were in the codebase but never hooked up with any actual type in the stdlib so the user were never able to create them. Instead of deleting it, I am slightly modifying the existing implementation so that it is now allowed in user code as `Tuple<T0,T1,...Tn>`.